File: server/app/schemas/health.py
"""
AgriSense AI — Health Analysis Schemas
=========================================
Pydantic schemas for crop health monitoring.
"""

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class HealthAnalysisRequest(BaseModel):
    """Schema for requesting crop health analysis."""

    plot_id: uuid.UUID
    crop_id: uuid.UUID | None = None
    notes: str | None = Field(default=None, max_length=500)
    # Image is uploaded as multipart form data, not in JSON body


class HealthAnalysisResponse(BaseModel):
    """Schema for health analysis results."""

    id: uuid.UUID
    plot_id: uuid.UUID
    crop_id: uuid.UUID | None = None
    image_path: str
    upload_date: datetime
    health_score: float | None = None
    diagnosis: str | None = None
    confidence: float | None = None
    severity: str | None = None
    recommendation_text: str | None = None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class HealthHistoryResponse(BaseModel):
    """Paginated health analysis history."""

    records: list[HealthAnalysisResponse]
    total: int
    page: int
    page_size: int

File: client/src/types/health.ts
/**
 * AgriSense AI — Health Types
 * ==============================
 */

export type HealthSeverity = "none" | "low" | "medium" | "high" | "critical";
export type HealthStatus = "pending" | "processing" | "completed" | "failed";

export interface HealthRecord {
  id: string;
  plot_id: string;
  crop_id?: string | null;
  image_path: string;
  upload_date: string;
  ai_result?: string | null;
  health_score?: number | null;
  diagnosis?: string | null;
  confidence?: number | null;
  severity?: HealthSeverity | null;
  recommendation_text?: string | null;
  status: HealthStatus;
  created_at: string;
}

export interface HealthListResponse {
  records: HealthRecord[];
  total: number;
  page: number;
  page_size: number;
}

export interface HealthListParams {
  page?: number;
  page_size?: number;
  plot_id?: string;
}
